lib/retry.ts

```diff
@@ -0,0 +1,101 @@
+export function operation(
+  options: OperationOptions,
+  fn: (currentAttempt: number) => void
+) {
+  return new RetryOperation(timeouts(options), fn);
+}
+
+export function timeouts(options: OperationOptions) {
+  if (options.minTimeout > options.maxTimeout) {
+    throw new Error("minTimeout is greater than maxTimeout");
+  }
+
+  const timeouts: number[] = [];
+  for (let i = 0; i < options.retries; i++) {
+    timeouts.push(createTimeout(i, options));
+  }
+
+  // sort the array numerically ascending
+  timeouts.sort(function(a, b) {
+    return a - b;
+  });
+
+  return timeouts;
+}
+
+function createTimeout(
+  attempt: number,
+  opts: Required<CreateTimeoutOptions>
+): number {
+  const random = opts.randomize ? Math.random() + 1 : 1;
+
+  let timeout = Math.round(
+    random * opts.minTimeout * Math.pow(opts.factor, attempt)
+  );
+  timeout = Math.min(timeout, opts.maxTimeout);
+
+  return timeout;
+}
+export interface OperationOptions extends CreateTimeoutOptions {
+  /**
+   * Whether to [unref](https://nodejs.org/api/timers.html#timers_unref) the setTimeout's.
+   * @default false
+   */
+  unref?: boolean;
+  /**
+   * The maximum amount of times to retry the operation.
+   * @default 10
+   */
+  retries: number;
```

Comment:
Where are the defaults defined? Some of the numbers look different here
https://github.com/zalando-incubator/perron/blob/800cdf824fb353f08244ced6db5059cc072cf6da/lib/client.ts#L138-L142

Reply:
We use different defaults.

Reply:
Updated the comments to match our defaults.